### PR TITLE
test fcx2rowvec

### LINKED ISSUE #74: fcx2rowvec - tests
Create unit tests for the function fcx2rowvec.m

### PR DESCRIPTION
**Pull request by j-emberton**
closes #74 

This PR develops unit tests for fcx2rowvec.m.

This function flattens multidimensional arrays into rows, however it does it with Matlab column major formatting.

All tests pass

### CODE REVIEW

**Review by isaacaka — APPROVED** [review]
Looks good, all tests pass

**Comment by dorchard** [comment]
I've done some simplification here: https://github.com/Cambridge-ICCS/ONEFlux/pull/80/ but the tests themselves look good to me.

File: tests/unit_tests/test_ustar_cp/test_fcx2rowvec.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    Test MATLAB's fcx2rowvec function with various matrices and arrays, verifying reshaping.
+    """
+    # Call MATLAB function
+    result = matlab_engine.fcx2rowvec(input_data)
+
+    # Extract values from MATLAB result
+    if isinstance(result, float):  # Scalar result from MATLAB
+        result_list = [result]
+    elif isinstance(result, matlab.double):  # Array result from MATLAB
+        result_list = list(result[0]) if len(result) > 0 else []
```
Comment (dorchard) [inline tests/unit_tests/test_ustar_cp/test_fcx2rowvec.py]: `len(result) > 0` will always be true.

File: tests/unit_tests/test_ustar_cp/test_fcx2rowvec.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    else:
+        raise TypeError(f"Unexpected result type: {type(result)}")
+
+    # Compare expected and actual results
+    assert len(result_list) == len(expected), "Result length mismatch"
+    for idx, exp_val in enumerate(expected):
+        if np.isnan(exp_val):
+            assert np.isnan(result_list[idx]), f"Expected NaN, got {result_list[idx]} at index {idx}"
+        else:
+            assert result_list[idx] == exp_val, f"Expected {exp_val}, got {result_list[idx]} at index {idx}"
```
Comment (dorchard) [inline tests/unit_tests/test_ustar_cp/test_fcx2rowvec.py]: I think we can simplify this away with a slightly more powerful `compare_matlab_arrays`. See https://github.com/Cambridge-ICCS/ONEFlux/pull/80/

**Review by dorchard — COMMENTED** [review]
Tests look good apart from a need to potentially simplify.

**Pull request merged**